### ISSUE TITLE
ShadowSocks security issue

### ISSUE THREAD

**Issue by sailosha** [Jigsaw-Code/outline-server#21]
It seems like China has already developed sniff & tracking machine learning to track ss data transfer. [Link for the research. ](http://ieeexplore.ieee.org/document/8048116/?reload=true#full-text-section)
They DID give the warning for the people who own ss VPN servers. 

If we didn't change ss protocol, it wouldn't work so well in the long term.

**Comment by fortuna** [comment]
We have found no evidence that Shadowsocks can be detected based on protocol fingerprinting alone, so I'll close this bug at this time. We can open a new issue if there's concrete evidence that Shadowsocks has been broken.

**Comment by mashirozx** [comment]
I do agree that Shadowsocks maybe not secure enough in China. Actually, we have known lots of Shadowsocks server had been blocked accurately in China. The truth is, for example, most Vultr's IP in Tokyo is not accessible in China now  due to the abuse of Shadowsocks.

**Comment by mashirozx** [comment]
Without any evidence, but we believe this [ShadowsocksR](https://github.com/shadowsocksr-backup/shadowsocksr) would be more secure in China.

**Comment by sailosha** [comment]
@mashirozx I do agree with you, but SSR is not open source, and the original creator is against GPL. Well, it is not important, the major issue is: not open souce, creator delete all the code. It is always hard to balance the payload for the vpn package vs speed. I had ssr server on google cloud, it works so well XD

**Comment by mashirozx** [comment]
@sailosha I'm pretty worry that this tool is too easy to use for a normal person, they can even finish all installation just by one click. This kind of abusing unsecure proxy may lead to a wild ban of IDC's IP addresses in China, it may be damaging...

**Comment by laosb** [comment]
@mashirozx Then they ban it. You always got some solutions to do. Even they entirely cut off China Mainland from Internet.

**Comment by mashirozx** [comment]
@laosb I guess it is **NOT YOU** who has the ability to get the solution. If that's right, just shut up.

**Comment by studentmain** [comment]
So, why not try v2ray.

**Comment by laosb** [comment]
@mashirozx But not you either, needless to guess. Just calm down and really put your mind to think about it, man. Being angry won't make you unblocked in any sense.

**Comment by mashirozx** [comment]
@laosb Do you really understand what I'm talking? I just worry about using Shadowsocks incorrectly on a grand scale may lead to an entirely IP block of these cheap IDCs. Yeah, you have a load of solutions... without using a foreign VPS/IP?

**Comment by paperbag** [comment]
>It seems like China has already developed sniff & tracking machine learning to track ss data transfer. Link for the research. 

That's all we known. But this paper is absolutely wrong.

I would like refer fortuna's comment posted last year.

>The paper says:
>
>>Like most principles of proxy software, Shadowsocks firstly establish a SSH [8] based encrypted channel with servers which are outside the firewall
>
>That's simply wrong: Shadowsocks does not use SSH. Did they test the right thing? SSH is a lot easier to identify.

https://github.com/shadowsocks/shadowsocks/issues/988#issuecomment-336508578

>Without any evidence, but we believe this ShadowsocksR would be more secure in China.

In theoretically, there is no traffic can't be analysis. ShadowsocksR also can be  entropy attack, according to [Seeing through network-protocol obfuscation](http://pages.cs.wisc.edu/~liangw/pub/ccsfp653-wangA.pdf).

**Comment by laosb** [comment]
@mashirozx if they do want to do this, then this factor is really not worth considering. Complete banning is basically entirely different from selective banning, and politic factors are greater when compared to tech factors.

**Comment by justin223** [comment]
Guys, please add **obfuscation** to server and client implementations, as it's easy to be detected by GFW and the whole IP segment of VPS vendor will be blocked by GFW.

I've tried several vendor (Vultr, DO, Scaleway...) **without obfuscation** and only a few days working before my IP been blocked. Yes, it was shadowsocks that time that I deployed on my servers. Then I deploy **shadowsocks with simple-obfs plugin**, **both server side and client side**, everything goes well till now.

Please check on **shadowsocks-libev** and **simple-obfs** repos to find solution to this.
Sever side:
[shadowsocks-libev](https://github.com/shadowsocks/shadowsocks-libev)
[simple-obfs](https://github.com/shadowsocks/simple-obfs)

Client side:
[simple-obfs-android](https://github.com/shadowsocks/simple-obfs-android)
[simple-obfs](https://github.com/shadowsocks/simple-obfs)


****IT'S** URGENT** as the more users, the easier it is to be detected and blocked.
